src/sqlancer/yugabyte/ysql/YSQLProvider.java

```diff
@@ -0,0 +1,366 @@
+        if (globalState.getDbmsSpecificOptions().getTestOracleFactory().stream()
+                .anyMatch((o) -> o == YSQLOracleFactory.PQS)) {
+            generateOnlyKnown = true;
+        }
+
+        username = globalState.getOptions().getUserName();
+        password = globalState.getOptions().getPassword();
+        host = globalState.getOptions().getHost();
+        port = globalState.getOptions().getPort();
+        entryPath = "/yugabyte";
+        entryURL = globalState.getDbmsSpecificOptions().connectionURL;
+        String entryDatabaseName = entryPath.substring(1);
+        databaseName = globalState.getDatabaseName();
+
+        try {
+            URI uri = new URI(entryURL);
+            String userInfoURI = uri.getUserInfo();
+            String pathURI = uri.getPath();
+            if (userInfoURI != null) {
+                // username and password specified in URL take precedence
+                if (userInfoURI.contains(":")) {
+                    String[] userInfo = userInfoURI.split(":", 2);
+                    username = userInfo[0];
+                    password = userInfo[1];
+                } else {
+                    username = userInfoURI;
+                    password = null;
+                }
+                int userInfoIndex = entryURL.indexOf(userInfoURI);
+                String preUserInfo = entryURL.substring(0, userInfoIndex);
+                String postUserInfo = entryURL.substring(userInfoIndex + userInfoURI.length() + 1);
+                entryURL = preUserInfo + postUserInfo;
+            }
+            if (pathURI != null) {
+                entryPath = pathURI;
+            }
+            if (host == null) {
+                host = uri.getHost();
+            }
+            if (port == MainOptions.NO_SET_PORT) {
+                port = uri.getPort();
+            }
+            entryURL = String.format("jdbc:yugabytedb://%s:%d/%s", host, port, entryDatabaseName);
+        } catch (URISyntaxException e) {
+            throw new AssertionError(e);
+        }
+
+        createDatabaseSync(globalState, entryDatabaseName);
+
+        int databaseIndex = entryURL.indexOf("/" + entryDatabaseName) + 1;
+        String preDatabaseName = entryURL.substring(0, databaseIndex);
+        String postDatabaseName = entryURL.substring(databaseIndex + entryDatabaseName.length());
+        testURL = preDatabaseName + databaseName + postDatabaseName;
+        globalState.getState().logStatement(String.format("\\c %s;", databaseName));
+
+        return new SQLConnection(createConnectionSafely(testURL, username, password));
+    }
+
+    @Override
+    public String getDBMSName() {
+        return "ysql";
+    }
+
+    // for some reason yugabyte unable to create few databases simultaneously
+    private void createDatabaseSync(YSQLGlobalState globalState, String entryDatabaseName) throws SQLException {
+        synchronized (CREATION_LOCK) {
+            exceptionLessSleep(5000);
```

Comment:
You introduced some timeouts here and at other places. I think it would be ideal to introduce some named constants, perhaps in `YSQLCommon`, which might make it easier to debug potential performance problems later on.

Reply:
Added comment